### ISSUE TITLE
Debian Server Installation Difficulties - Support Request rather than a Bug Report

### ISSUE THREAD

**Issue by Whyglobaleyes** [automatic-ripping-machine/automatic-ripping-machine#462]
### Environment
1. `cat /etc/lsb-release` ==> `No such file or directory` {which I don't understand at all)  
2. `uname -a` ==> `Linux theNas 4.19.0-18-686-pae #1 SMP Debian 4.19.208-1 (2021-09-29) i686 GNU/Linux`
3. ARM version ==> from 2 days ago (2021.11.30)

### Description
1- SETTING: I had ARM up & running on another machine a few years ago and loved it but recently I did (a) a fresh install of Debian 10 Server edition (b) install of ARM using the attended install method under "Alternative Auto Install Script For OpenMediaVault/Debian" 
2. OUTCOME ==> ARM doesn't launch when I insert a CD. 

### Log file & steps tried so far 
1  There's no log file in /home/arm/ 
2 I tried the manual install steps under the ubuntu instructions and .... 
3 `sudo apt install makemkv-bin makemkv-oss` failed and returned ... 
```
E: Unable to locate package makemkv-bin
E: Unable to locate package makemkv-oss
```
==> I installed both from https://forum.makemkv.com/forum/viewtopic.php?f=3&t=224 
4. `apt install avahi-daemon` installed packages but none of the others needed to
5. I tried `tail -f /var/log/syslog` ==> I got an error along the lines of "No module named Flask"]
6. I manually installed = `pip3 install Flask` ... but I wonder if it needs to be installed in a particular location/ environment??
7. re-ran the Auto Install script (unattended mode) // reboot :: actually done this several times 
8. I've tried watch htop when I insert a CD and I see a few processes start = (a) /lib/systemd/systemd --user (b) (sd-pam)  
9. The error's changed now and the most important bit seems to be "No module named 'flask_sqlalchemy'" but the output of `tail -f /var/log/syslog` is: 
```
Dec  2 17:47:02 theNas systemd[1]: armui.service: Failed with result 'exit-code'.
Dec  2 17:47:06 theNas systemd[1]: armui.service: Service RestartSec=3s expired, scheduling restart.
Dec  2 17:47:06 theNas systemd[1]: armui.service: Scheduled restart job, restart counter is at 2740.
Dec  2 17:47:06 theNas systemd[1]: Stopped Arm service.
Dec  2 17:47:06 theNas systemd[1]: Started Arm service.
Dec  2 17:47:07 theNas python3[9620]: Traceback (most recent call last):
Dec  2 17:47:07 theNas python3[9620]:   File "/opt/arm/arm/runui.py", line 5, in <module>
Dec  2 17:47:07 theNas python3[9620]:     from arm.ui import app  # noqa E402
Dec  2 17:47:07 theNas python3[9620]:   File "/opt/arm/arm/../arm/ui/__init__.py", line 2, in <module>
Dec  2 17:47:07 theNas python3[9620]:     from flask_sqlalchemy import SQLAlchemy
Dec  2 17:47:07 theNas python3[9620]: ModuleNotFoundError: No module named 'flask_sqlalchemy'
Dec  2 17:47:07 theNas systemd[1]: armui.service: Main process exited, code=exited, status=1/FAILURE
Dec  2 17:47:07 theNas systemd[1]: armui.service: Failed with result 'exit-code'.
```
I love ARM but I'm a bit, actually way beyond my depth trying to fix the installation :: I can't give a logical reason for it but I wonder if the problem is something to do with my server being (a) Debian and (b) a headless install (I ssh into it) :: the last time I had ARM was on a modern laptop with a full GUI or (c) having two drives (I've tried using both).

Any help will be very gratefully received
bw
Alex

**Comment by 1337-server** [comment]
Have you tried `pip3 install flask_sqlalchemy` ?

**Comment by Whyglobaleyes** [comment]
TLDR = installation abandoned: that machine had a noble death :: it's a Compaq Pressario circa 1999 with a 32bit chip :: I'll try ARM on a slightly more up to date - but 64bit -  box ;o) and when I'm really board I'll see if the Compaq can be resurrected, I'm fond of it.

@1337-server :: Thanks for the suggestion :: It got me thinking and FWIW I managed to do a little snooping around before the poor thing gave up: 
1. I ran `pip3 list` on the box and had a look at `required.txt` from this repo ... 
2. I only had 3 of the required python programmes = pycurl :: requests :: urllib3
3. My inference is that I had 22 python packages to download ==> perhaps that's something to do with it being a 32bit os?? Anyone have any other ideas about it?

bw 
Alex

_**list of missing python programmes:** apprise :: bcrypt :: discid :: flake8 :: flask :: flask-login :: flask-migrate :: flask-sqlalchemy :: flask-WTF :: musicbrainzngs :: netifaces :: omdb :: psutil :: pushbullet.py :: pydvdid :: pyfttt :: python-pushover :: pyudev :: pyyaml :: robobrowser :: tinydownload :: xmltodict_

**Comment by flammableliquids** [comment]
@1337-server 

I was doing a fresh install on a fresh OS. (Bare metal, not docker), and I hit a failure to 

`<uname>@armcache:~$ sudo systemctl status armui
● armui.service - Arm service
     Loaded: loaded (/etc/systemd/system/armui.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-03-12 21:20:29 AEDT; 459ms ago
    Process: 14205 ExecStartPre=/bin/bash -xc /usr/bin/systemctl is-active --quiet armui.service && exit 1 || exit 0 (code=exited, status=0/SUCCESS)
    Process: 14207 ExecStart=/usr/bin/python3 /opt/arm/arm/runui.py (code=exited, status=1/FAILURE)
   Main PID: 14207 (code=exited, status=1/FAILURE)
`

the log didn't give me anything useful:

`$ tail -f /var/log/syslog
Mar 12 21:16:25 armcache systemd[1]: Starting Arm service...
Mar 12 21:16:25 armcache systemd[1]: Started Arm service.
Mar 12 21:16:25 armcache systemd[1]: armui.service: Main process exited, code=exited, status=1/FAILURE
Mar 12 21:16:25 armcache systemd[1]: armui.service: Failed with result 'exit-code'.
Mar 12 21:16:28 armcache systemd[1]: armui.service: Scheduled restart job, restart counter is at 83.
Mar 12 21:16:28 armcache systemd[1]: Stopped Arm service.
Mar 12 21:16:28 armcache systemd[1]: Starting Arm service...
Mar 12 21:16:28 armcache systemd[1]: Started Arm service.
Mar 12 21:16:28 armcache systemd[1]: armui.service: Main process exited, code=exited, status=1/FAILURE
Mar 12 21:16:28 armcache systemd[1]: armui.service: Failed with result 'exit-code'.
Mar 12 21:16:31 armcache systemd[1]: armui.service: Scheduled restart job, restart counter is at 84.
Mar 12 21:16:31 armcache systemd[1]: Stopped Arm service.
`

I couldn't narrow down the problem until I manually executed the service, where I found that python packages were missing:

` /usr/bin/python3 /opt/arm/arm/runui.py
Traceback (most recent call last):
  File "/opt/arm/arm/runui.py", line 8, in <module>
    import arm.config.config as cfg  # noqa E402
  File "/opt/arm/arm/../arm/__init__.py", line 2, in <module>
    import arm.ripper
  File "/opt/arm/arm/../arm/ripper/__init__.py", line 3, in <module>
    from arm.ripper import logger, utils, makemkv, handbrake, identify  # noqa F401
  File "/opt/arm/arm/../arm/ripper/utils.py", line 16, in <module>
    import bcrypt
`

The above list of python modules were missing, but also including flask_cors and  prettytable (which had to be installed as arm user)

Still haven't gotten it to run yet.
